dm/dm/master/scheduler/scheduler.go

```diff
@@ -2193,6 +2230,25 @@ func (s *Scheduler) tryBoundForSource(source string) (bool, error) {
 	return true, nil
 }
 
+// tryBoundSourceToWorker tries to bound a source to the specified worker.
+// returns (true, nil) after bounded.
+func (s *Scheduler) tryBoundSourceToWorker(source string, w *Worker) (bool, error) {
+	boundedSource := ""
+	for s, worker := range s.bounds {
+		if worker.baseInfo.Name == w.baseInfo.Name {
+			boundedSource = s
+			break
+		}
+	}
+
+	if boundedSource != "" {
+		return false, nil
+	}
```

Comment:
We can use w.Stage() and w.Bound() to know if a worker is bounded.

BTW, we have checked w.stage == WorkerFree in caller, and caller holds a lock so there should not be another goroutine modifying scheduler status, can we remove this boundedSource check?

Reply:
here we can call boundSourceToWorker directly because of the pre-check

Reply:
this method is only called by AddSourceCfgWithWorker, and i can remove it and can call boundSourceToWorker directly